### PR TITLE
Add speech-to-text typos transformation

### PR DESCRIPTION
**Pull request by goyalpramod**
## Description

add speech-to-text typos transformation to transformation functions catalog
used homophones from the following [link](https://www.cs.cmu.edu/afs/cs/project/ai-repository/ai/areas/speech/database/homofonz/)
## Related Issue

#1566 
## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [x] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

**Comment by goyalpramod** [comment]
@luca-martial kindly have a look and let me know any changes you would like me to make. I did not make any changes in perturbation.py because I was uncertain if that needed to be changed. 
I have implemented the solution in a word level. I tried thinking of a way in which this could be solved in the phonetic level but could not come up with anything fruitful.

**Comment by goyalpramod** [comment]
@luca-martial I apologize for tagging you again, you may have missed this PR. I understand you have a busy schedule, but could you kindly review it whenever you get the time?

**Comment by kevinmessiaen** [comment]
Hello, thanks for your contribution, took a quick look and really love this transformatio! I'll be reviewing this PR ASAP

File: giskard/scanner/robustness/text_transformations.py [hunk 1]
```diff
@@ -122,6 +122,57 @@ def _random_key_typo(self, char):
...
         return char
 
+class TextFromSpeechTypoTransformation(TextTransformation):
+    name = "Add text from speech typos"
+    
+    def __init__(self, column, rate=0.05, min_length=10, rng_seed=1729):
+        super().__init__(column)
+        from .entity_swap import speech_typos
+
+        self.rate = rate
```
Comment (kevinmessiaen) [inline giskard/scanner/robustness/text_transformations.py:132]: It seems that we are not using this rate and instead make typo to all words matched in the dict

File: giskard/scanner/robustness/text_transformations.py [hunk 2]
```diff
@@ -122,6 +122,57 @@ def _random_key_typo(self, char):
...
+        self.min_length = min_length
+        self._word_typos = speech_typos
+        self.rng = np.random.default_rng(seed=rng_seed)
+
+    def make_perturbation(self, x):
+        # Skip if the text is too short
+        if len(x) < self.min_length:
+            return x
+
```
Comment (kevinmessiaen) [inline giskard/scanner/robustness/text_transformations.py:141]: I guess we would want to skip for languages other than English too

```suggestion
        # Skip if language isn't supported
        if row["language__gsk__meta"] not 'en':
            return x 
```

**Review by kevinmessiaen — CHANGES_REQUESTED** [review]
Thanks for the really nice PR, I just left a few comment

File: giskard/scanner/robustness/text_transformations.py (hunk 1 above)
Comment (goyalpramod) [reply giskard/scanner/robustness/text_transformations.py]: I have made the changes as requested. Do let me know if you would like me to add or change anything else.

File: giskard/scanner/robustness/text_transformations.py (hunk 2 above)
Comment (kevinmessiaen) [reply giskard/scanner/robustness/text_transformations.py]: Just fixed compilation

File: giskard/scanner/robustness/text_transformations.py (hunk 1 above)
Comment (kevinmessiaen) [reply giskard/scanner/robustness/text_transformations.py]: Thanks, it's good!

**Review by kevinmessiaen — APPROVED** [review]
LGTM, thanks for your contribution 🎉

**Pull request merged**